src/ApiResourceExampleDocument.js

```diff
@@ -317,6 +328,28 @@ export class ApiResourceExampleDocument extends AmfHelperMixin(LitElement) {
     }));
   }
 
+  get maxHeight() {
+    return this._maxHeight;
+  }
+
+  /**
+   * @param {string} value
+   */
+  set maxHeight(value) {
+    const old = this._maxHeight;
+    if (old === value) {
+      return;
+    }
+    this._maxHeight = value;
+    this.requestUpdate('maxHeight', old);
+    this.dispatchEvent(new CustomEvent('max-height-changed', {
```

Comment:
Dispatching an event in a setter is not recommended. There's no way to determine how the change was performed. Events should only be dispatched when the state changes internally and from the place where this change happened.

Reply:
Done